ext/oci8/oci8.c

```diff
@@ -1891,9 +1890,14 @@ static php_oci_spool *php_oci_create_spool(char *username, int username_len, cha
 		goto exit_create_spool;
 	}
 
-/* Disable RLB as we mostly have single-connection pools */
-	poolmode = OCI_SPC_NO_RLB | OCI_SPC_HOMOGENEOUS;
-
+  /* if CRED_EXT mode poolcreate set to hetero(OCI_DEFAULT) 
+   */
+  if (mode & PHP_OCI_CRED_EXT)
+    poolmode = OCI_DEFAULT;
+  else {
+    /* Disable RLB as we mostly have single-connection pools */
+    poolmode = OCI_SPC_NO_RLB | OCI_SPC_HOMOGENEOUS;
+  }
```

Comment:
nits

The same applies to other parts, but it is preferable to write them using `{}`.

```suggestion
  if (mode & PHP_OCI_CRED_EXT) {
    poolmode = OCI_DEFAULT;
  } else {
    /* Disable RLB as we mostly have single-connection pools */
    poolmode = OCI_SPC_NO_RLB | OCI_SPC_HOMOGENEOUS;
  }
```